initialization/initialization.go

```diff
@@ -203,38 +237,29 @@ func (init *Initializer) Reset() error {
 	return nil
 }
 
-func (init *Initializer) Started() (bool, error) {
-	numLabelsWritten, err := init.diskState.NumLabelsWritten()
-	if err != nil {
-		return false, err
+func (init *Initializer) Status() Status {
+	if init.isInitializing() {
+		return StatusInitializing
 	}
 
-	return numLabelsWritten > 0, nil
-}
-
-func (init *Initializer) Completed() (bool, error) {
 	numLabelsWritten, err := init.diskState.NumLabelsWritten()
 	if err != nil {
-		return false, err
+		return StatusError
 	}
 
 	target := uint64(init.opts.NumUnits) * uint64(init.cfg.LabelsPerUnit)
-	return numLabelsWritten == target, nil
-}
-
-func (init *Initializer) VerifyStarted() error {
-	ok, err := init.Started()
-	if err != nil {
-		return err
+	if numLabelsWritten == target {
+		return StatusCompleted
 	}
-	if !ok {
-		return shared.ErrInitNotStarted
+
+	if numLabelsWritten > 0 {
+		return StatusStarted
 	}
 
-	return nil
+	return StatusNotStarted
```

Comment:
Given these conditions `StatusStarted` means "already produced something" and `StatusNodStarted` means "not produced anything yet (but might be started already)".

Reply:
`StatusNotStarted` only means "not produced anything yet (but might be started already)" if `Initialize()` is called between the check of `initialize` at the top of this method and this return.

Here again I think the removal of `initialize` and using `TryLock()` might resolve the issues with the `Status()` method.